src/transaction/components/TransactionProvider.tsx

```diff
@@ -76,6 +76,10 @@ export function TransactionProvider({
     ? TRANSACTION_TYPE_CALLS
     : TRANSACTION_TYPE_CONTRACTS;
 
+  const [prevStatusEmitted, setPrevStatusEmitted] = useState(
```

Comment:
nit: can we use `previousStatus, setPreviousStatus` instead

Reply:
@abcrane123 this approach adds more complexity to our logic, and makes it harder to mantain later.

What we should do instead, is add a persistant value inside statusData that keep info regarding the pending part.

With @alessey we are actually exploring this way of thinking for Swap and make sure become the new standard https://github.com/coinbase/onchainkit/pull/1237

Make sure to sync with @alessey and follow the same approach.